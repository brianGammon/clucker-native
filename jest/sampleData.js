const sampleData1 = {
  userSettings: {
    key: 'user1',
    data: {
      currentFlockId: 'flock1',
    },
  },
  flocks: {
    data: {
      flock1: {
        name: 'Test Flock 1',
        ownedBy: 'user1',
      },
      flock2: {
        name: 'Test Flock 2',
        ownedBy: 'user2',
      },
    },
  },
  chickens: {
    data: {
      chicken1: {
        name: 'The First Chicken',
      },
      chicken2: {
        name: 'The Second Chicken',
      },
    },
  },
  eggs: {
    data: {
      egg1: {
        chickenId: 'chicken1',
        chickenName: 'Bossie',
        damaged: false,
        date: '2018-08-01',
        modified: '2018-01-12T18:05:54.806Z',
        notes:
          'This was the record setting egg from May 2017, real egg was 66.1g on this day',
        userId: 'MVQUEcoTaVUhUdo2ZS0zsIE6tn93',
        weight: 88.1,
      },
      egg2: {
        chickenId: 'chicken2',
        chickenName: 'Baby',
        damaged: false,
        date: '2017-09-24',
        modified: '2017-10-12T18:05:54.806Z',
        userId: 'MVQUEcoTaVUhUdo2ZS0zsIE6tn93',
        weight: 88.1,
      },
      egg3: {
        chickenId: 'chicken1',
        chickenName: 'Bossie',
        damaged: false,
        date: '2018-08-02',
        modified: '2018-08-02T18:05:54.806Z',
        notes: '',
        userId: 'MVQUEcoTaVUhUdo2ZS0zsIE6tn93',
        weight: 68.1,
      },
    },
  },
};

export default sampleData1;
