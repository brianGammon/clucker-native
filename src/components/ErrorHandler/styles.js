/* @flow */
import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  modalContainer: {
    flex: 1,
    justifyContent: 'center',
  },
  container: {
    padding: 15,
    alignSelf: 'center',
    width: 300,
    backgroundColor: '#d9534f',
    borderRadius: 10,
  },
  helpTextContainer: {
    marginVertical: 20,
  },
  helpTextItem: {
    marginBottom: 4,
  },
  buttonsContainer: {
    flexDirection: 'row',
    justifyContent: 'space-evenly',
    marginTop: 10,
  },
  buttonText: {
    paddingLeft: 0,
  },
  accordianHeader: {
    backgroundColor: '#d9534f',
  },
  accordianContent: {
    backgroundColor: '#ea7f7c',
  },
});
