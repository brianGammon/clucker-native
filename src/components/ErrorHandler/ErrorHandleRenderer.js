/* @flow */
import React from 'react';
import {
  View, Text, Button, Icon, Accordion, Content, H3,
} from 'native-base';
import styles from './styles';

type Props = {
  error: string,
  uid: string,
  startListening: (uid: string) => void,
  clearErrors: () => void,
};

const ErrorHandlerRenderer = ({
  error,
  uid,
  startListening,
  clearErrors,
}: Props) => (
  <Content style={styles.container}>
    <H3>Sorry, something went wrong</H3>
    <View style={styles.helpTextContainer}>
      <Text style={styles.helpTextItem}>Things you can try:</Text>
      <Text style={styles.helpTextItem}>
        {`${'\u2022'} Make sure you have an internet connection, then tap Reload.`}
      </Text>
      <Text style={styles.helpTextItem}>
        {`${'\u2022'} Sign out in Settings, then sign in again.`}
      </Text>
      <View style={styles.buttonsContainer}>
        <Button bordered dark small onPress={() => startListening(uid)}>
          <Icon name="refresh" />
          <Text style={styles.buttonText}>Reload</Text>
        </Button>
        <Button bordered dark small onPress={() => clearErrors()}>
          <Icon name="close" />
          <Text style={styles.buttonText}>Dismiss</Text>
        </Button>
      </View>
    </View>
    <Accordion
      dataArray={[{ title: 'Error Details', content: error }]}
      headerStyle={styles.accordianHeader}
      contentStyle={styles.accordianContent}
      expanded={1}
    />
  </Content>
);

export default ErrorHandlerRenderer;
