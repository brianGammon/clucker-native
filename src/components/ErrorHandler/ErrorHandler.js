/* @flow */
import React from 'react';
import { connect } from 'react-redux';
import * as actions from '../../redux/actions';
import ErrorHandleRenderer from './ErrorHandleRenderer';
import { actionTypes, metaTypes } from '../../redux/constants';

type Props = {
  error: string,
  uid: string,
  startListening: (uid: string) => void,
  clearErrors: () => void,
};

const ErrorHandler = ({
  error, uid, startListening, clearErrors,
}: Props) => {
  if (!error) {
    return null;
  }
  return (
    <ErrorHandleRenderer
      error={error}
      uid={uid}
      startListening={startListening}
      clearErrors={clearErrors}
    />
  );
};

const mapStateToProps = ({
  auth: { user },
  chickens: { error: chickensError },
  eggs: { error: eggsError },
}) => ({
  error: chickensError || eggsError,
  uid: user && user.uid,
});

const mapDispatchToProps = dispatch => ({
  startListening: (userId) => {
    dispatch(actions.listenToChickens(userId));
    dispatch(actions.listenToEggs(userId));
  },
  clearErrors: () => {
    dispatch({
      type: actionTypes.CLEAR_ERROR,
      meta: { type: metaTypes.chickens },
    });
    dispatch({ type: actionTypes.CLEAR_ERROR, meta: { type: metaTypes.eggs } });
  },
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(ErrorHandler);
