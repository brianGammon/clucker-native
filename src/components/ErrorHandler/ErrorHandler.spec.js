import React from 'react';
import configureStore from 'redux-mock-store';
import sampleData from '../../../jest/test-data/full.2017-10-to-2018-05.json';
import ErrorHandler from '.';
import ErrorHandlerRenderer from './ErrorHandleRenderer';

describe('ErrorHandler', () => {
  const middlewares = [];
  const mockStore = configureStore(middlewares);

  test('Connected component renders with error message', () => {
    const store = mockStore({
      ...sampleData,
      chickens: { ...sampleData.chickens, error: 'Some error occurred' },
    });
    const wrapper = shallow(<ErrorHandler store={store} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Renderer component renders', () => {
    const wrapper = shallow(
      <ErrorHandlerRenderer
        error="Test Error Message"
        uid="user1"
        startListening={() => {}}
        clearErrors={() => {}}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
