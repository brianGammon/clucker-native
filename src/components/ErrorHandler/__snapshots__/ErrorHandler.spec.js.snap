// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorHandler Connected component renders with error message 1`] = `
<ErrorHandler
  clearErrors={[Function]}
  error="Some error occurred"
  startListening={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
  uid="user1"
/>
`;

exports[`ErrorHandler Renderer component renders 1`] = `
<Component
  animationType="fade"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  transparent={true}
  visible={true}
>
  <Styled(ViewNB)
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Styled(ViewNB)
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#d9534f",
          "borderRadius": 10,
          "padding": 15,
          "width": 300,
        }
      }
    >
      <Styled(H3)>
        Sorry, something went wrong
      </Styled(H3)>
      <Styled(ViewNB)
        style={
          Object {
            "marginVertical": 20,
          }
        }
      >
        <Styled(Text)
          style={
            Object {
              "marginBottom": 4,
            }
          }
        >
          Things you can try:
        </Styled(Text)>
        <Styled(Text)
          style={
            Object {
              "marginBottom": 4,
            }
          }
        >
          • Make sure you have an internet connection, then tap Reload.
        </Styled(Text)>
        <Styled(Text)
          style={
            Object {
              "marginBottom": 4,
            }
          }
        >
          • Sign out in Settings, then sign in again.
        </Styled(Text)>
        <Styled(ViewNB)
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-evenly",
              "marginTop": 10,
            }
          }
        >
          <Styled(Button)
            bordered={true}
            dark={true}
            onPress={[Function]}
            small={true}
          >
            <Styled(Icon)
              name="refresh"
            />
            <Styled(Text)
              style={
                Object {
                  "paddingLeft": 0,
                }
              }
            >
              Reload
            </Styled(Text)>
          </Styled(Button)>
          <Styled(Button)
            bordered={true}
            dark={true}
            onPress={[Function]}
            small={true}
          >
            <Styled(Icon)
              name="close"
            />
            <Styled(Text)
              style={
                Object {
                  "paddingLeft": 0,
                }
              }
            >
              Dismiss
            </Styled(Text)>
          </Styled(Button)>
        </Styled(ViewNB)>
      </Styled(ViewNB)>
      <Accordion
        contentStyle={
          Object {
            "backgroundColor": "#ea7f7c",
          }
        }
        dataArray={
          Array [
            Object {
              "content": "Test Error Message",
              "title": "Error Details",
            },
          ]
        }
        expanded={1}
        headerStyle={
          Object {
            "backgroundColor": "#d9534f",
          }
        }
      />
    </Styled(ViewNB)>
  </Styled(ViewNB)>
</Component>
`;
